collectfast/strategies/filesystem.py

```diff
@@ -11,3 +15,27 @@ def get_remote_file_hash(self, prefixed_path: str) -> Optional[str]:
             return self.get_local_file_hash(prefixed_path, self.remote_storage)
         except FileNotFoundError:
             return None
+
+
+class CachingFileSystemStrategy(
+        CachingHashStrategy[FileSystemStorage],
+        FileSystemStrategy,
+):
+    @lru_cache()
+    def get_local_file_hash(self, *args, **kwargs):
+        '''
+        caches the local file hash in memory so the hash is only computed once
+        per run and `post_copy_hook` method the does not duplicate work
+        '''
+        return super().get_local_file_hash(*args, **kwargs)
+    
+    def post_copy_hook(
+            self, path: str, prefixed_path: str, local_storage: Storage
+    ) -> None:
+        '''
+        sets the cached hash from the local file that was just copied to avoid
+        reading the remote file which may be stored on a networked file system
+        '''
```

Comment:
Should we call `super().post_copy_hook(...)` here?

Reply:
Yes